### PR TITLE
Irmin 1.0.0 packages

### PR DESCRIPTION
**Pull request by samoht**
Note: the descriptions are not final, I need to add `.descr` files in the upstream repo.

### CODE REVIEW

**Comment by yomimono** [comment]
@samoht the CI failures here look genuine - the `irmin-mirage` failure is a bit worrying.  I'm inclined to PR `mirageos-3-beta` to opam-repository as-is, without Irmin - of course we'll be happy to see it released ASAP, but it's got its own bunch of revdeps to solve which might be better served by having it in a separate PR to opam-repository.

**Comment by samoht** [comment]
@yomimono no problem! I will fix the revdeps today.

**Comment by yomimono** [comment]
Leaving this PR open so samoht can test the release, but a merge here would no longer be meaningful now that https://github.com/ocaml/opam-repository/pull/8526 is open and this repository will be removed shortly.

File: packages/irmin-mirage/irmin-mirage.1.0.0/opam [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "ocamlbuild" {build}
+  "ocamlfind"  {build}
+  "topkg"      {build & >= "0.7.8"}
+  "irmin"      {>= "1.0.0"}
+  "irmin-git"  {>= "1.0.0"}
+  "git-mirage" {>= "1.10.0"}
+  "ptime"
+  "mirage-kv-lwt"
+  "mirage-clock"
+  "result"
```
Comment (hannesm) [inline packages/irmin-mirage/irmin-mirage.1.0.0/opam]: this lacks `irmin-http` (or adjustment of its META), `mirage build` complains:

```
ocamlfind: Package `irmin-http' not found - required by `irmin-mirage'
Command exited with code 2.
run ['ocamlbuild' '-use-ocamlfind' '-classic-display' '-tags'
     'predicate(mirage_unix),warn(A-4-41-42-44),color(always),debug,bin_annot,strict_sequence,principal,safe_string'
     '-pkgs'
     'cohttp,conduit,conduit.mirage,decompress,functoria-runtime,io-page,io-page.unix,irmin,irmin-mirage,logs,lwt,magic-mime,mirage-clock,mirage-clock-unix,mirage-flow,mirage-flow-lwt,mirage-http,mirage-kv-lwt,mirage-logs,mirage-runtime,mirage-types,mirage-types-lwt,mirage-unix,nocrypto,nocrypto.lwt,omd,ptime,re,syndic,tcpip,tcpip.stack-socket,tcpip.tcpv4-socket,tcpip.udpv4-socket,tls,tls.mirage,tyxml,uuidm'
     '-cflags' '-g' '-lflags' '-g' '-tag-line' '<static*.*>: warn(-32-34)'
     '-X' '_build-ukvm' 'main.native']: exited with 10
```

Comment (samoht) [inline packages/irmin-mirage/irmin-mirage.1.0.0/opam]: Thanks, fixed in https://github.com/mirage/irmin/pull/412 I'll make a new release shortly

**Comment by samoht** [comment]
New release done, let's see what the CI says.

@hannesm do you have a branch somewhere with the porting to irmin.1.0 that you did? I can help if needed. (also how is the new API?)

**Comment by hannesm** [comment]
thx.  Canopy error:
```
File "canopy_store.ml", line 8, characters 29-49:
Error: This module is not a functor; it has type
       sig
         module Make = Irmin_mirage.Git.Mem.Make
         module KV = Irmin_mirage.Git.Mem.KV
       end
```

Canopy branch at https://github.com/hannesm/Canopy.git#irmin1 (on top of mirage3)

**Comment by samoht** [comment]
Ok I'm fixing Canopy.

**Comment by samoht** [comment]
Fixes pushed to https://github.com/Engil/Canopy/pull/79

I think I found a bug in `irmin tree` (the CLI tool), I'll check and make a new release if needed.

**Comment by samoht** [comment]
Closing in favour of https://github.com/ocaml/opam-repository/pull/8535

**Pull request closed without merging**